dist/dfu.js

```diff
@@ -382,7 +382,8 @@
                         case OPCODE.VALIDATE_FIRMWARE:
                             log('complete, reset...');
 
-                            controlChar.service.device.addEventListener("gattserverdisconnected", function() {
+                            server.device.addEventListener("gattserverdisconnected", function() {
```

Comment:
Was this the fix for your error as I intended to remove ```server``` from the chars object as it was a hack.

If so, I'm surprised that ```controlChar.service.device``` isn't the same object as ```server.device```